msdk-io/msdk-io-netcdf/src/main/java/io/github/msdk/io/netcdf/NetCDFMsScan.java

```diff
@@ -127,15 +127,7 @@ public NetCDFMsScan(Integer scanNumber, int[] scanStartPositions, float[] scanRe
   public Integer getNumberOfDataPoints() {
     if (numOfDataPoints == null) {
       final Integer scanIndex = getScanIndex();
-      try {
-        final int scanStartPosition[] = {scanStartPositions[scanIndex]};
-        final int scanLength[] =
-            {scanStartPositions[scanIndex + 1] - scanStartPositions[scanIndex]};
-        final Array massValueArray = massValueVariable.read(scanStartPosition, scanLength);
-        numOfDataPoints = massValueArray.getShape()[0];
-      } catch (IOException | InvalidRangeException e) {
-        throw new MSDKRuntimeException(e);
-      }
+      numOfDataPoints = scanStartPositions[scanIndex + 1] - scanStartPositions[scanIndex];
```

Comment:
No.
`scanIndex + 1` = `scanNumber`, and the value of `scanNumber` for the last scan would be `totalScans`.
`scanStartPositions` is of length `totalScans + 1`.